cmd/exit_sign.go

```diff
@@ -45,13 +47,29 @@ func newSubmitPartialExitCmd(runFunc func(context.Context, exitConfig) error) *c
 		{lockFilePath, false},
 		{validatorKeysDir, false},
 		{exitEpoch, false},
-		{validatorPubkey, true},
-		{beaconNodeURL, true},
+		{validatorPubkey, false},
+		{validatorIndex, false},
+		{beaconNodeEndpoints, true},
 		{beaconNodeTimeout, false},
 	})
 
 	bindLogFlags(cmd.Flags(), &config.Log)
 
+	wrapPreRunE(cmd, func(cmd *cobra.Command, _ []string) error {
+		valIdxPresent := cmd.Flags().Lookup(validatorIndex.String()).Changed
+		valPubkPresent := cmd.Flags().Lookup(validatorPubkey.String()).Changed
+
+		if strings.TrimSpace(config.ValidatorPubkey) == "" && !valIdxPresent {
+			//nolint:revive // we use our own version of the errors package.
+			return errors.New(fmt.Sprintf("%s or %s must be specified.", validatorIndex.String(), validatorPubkey.String()))
```

Comment:
Neat: prefix with "either".

Reply:
Thinking about it more, this error is wrong: the user must _at least_ provide X or Y, or both, but never none.

Thanks for pointing this out!